### PR TITLE
Allow creation of annotated tags

### PR DESCRIPTION
**Pull request by fsmiamoto**
## What
Allow the creation of annotated tags.
This implementation only supports adding a message to the tag but it could be extended in the future.

## Why
This was requested in #1381 as currently we only support creating the so called *lightweight tags*

### CODE REVIEW

**Comment by fsmiamoto** [comment]
@jesseduffield This PR still has some rough edges but let me know what you think of the general approach :)

File: pkg/gui/commits_panel.go [hunk 1]
```diff
@@ -571,25 +571,71 @@ func (gui *Gui) handleSquashAllAboveFixupCommits() error {
...
+				return gui.handleCreateLightweightTag(commitSha)
+			}},
+		{
+			displayString: gui.Tr.AnnotatedTag,
+			onPress: func() error {
+				return gui.handleCreateAnnotatedTag(commitSha)
+			}},
+	}
+
+	return gui.createMenu(gui.Tr.TagMenuTitle, items, createMenuOptions{showCancel: false})
```
Comment (jesseduffield) [inline pkg/gui/commits_panel.go:596]: I would go `showCancel: true` just because it's harmless to do so

File: pkg/i18n/english.go [hunk 2]
```diff
@@ -889,7 +894,11 @@ func englishTranslationSet() TranslationSet {
 		SetUpstreamMessage:                  "Are you sure you want to set the upstream branch of '{{.checkedOut}}' to '{{.selected}}'",
 		LcEditRemote:                        "edit remote",
 		LcTagCommit:                         "tag commit",
+		TagMenuTitle:                        "create tag",
```
Comment (jesseduffield) [inline pkg/i18n/english.go:897]: menu titles should use 'Sentence case' so this should be 'Create tag'

File: pkg/i18n/english.go [hunk 3]
```diff
@@ -889,7 +894,11 @@ func englishTranslationSet() TranslationSet {
...
 		TagNameTitle:                        "Tag name:",
+		TagMessageTitle:                     "Tag message: ",
+		AnnotatedTag:                        "annotated tag",
+		LightweightTag:                      "lightweight tag",
```
Comment (jesseduffield) [inline pkg/i18n/english.go:901]: conventionally we prefix lowercase fields with 'Lc' (see line 902 below)

**Review by jesseduffield — CHANGES_REQUESTED** [review]
nice work! Couple small things

**Comment by fsmiamoto** [comment]
Took the opportunity to give some updates to the docs on Integration Tests as the commands there were not working as expected.

Not sure if what I did to fix the tests is the right thing but here's what I've done:
- Run the Integration Test TUI with `go run ./test/lazyintegration/main.go`
- Re-record the tests related to tags, trying/guessing what was done at those tests - not sure if there's a good way to visualize the actions that were there before.

Although these tests are slow by their own nature, I've really liked the general approach! 
Some things are not quite there in the Integration Test TUI - the test list didn't scroll properly for me - so I might give some help there afterwards.

**Comment by fsmiamoto** [comment]
Also, not sure why the linting is failing on me :sweat_smile: 
Maybe a different `gofmt`version on the CI? I'm not able to reproduce it locally.

**Comment by fsmiamoto** [comment]
Linting problems were caused by Go 1.17, fixed at #1479.

**Comment by fsmiamoto** [comment]
Fixed the `pushTag` by adding an extra `Enter` input. Should be good to go now :)

**Comment by jesseduffield** [comment]
Just tested and it works fine. Nice work!

**Pull request merged**